metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/HashFunctionsTest.java

```diff
@@ -0,0 +1,169 @@
+
+  @Test(expected = IllegalArgumentException.class)
+  public void nullArgumentListShouldThrowException() throws Exception {
+    hash.apply(null);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void emptyArgumentListShouldThrowException() throws Exception {
+    hash.apply(Collections.emptyList());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void singleArgumentListShouldThrowException() throws Exception {
+    hash.apply(Collections.singletonList("some value."));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void argumentListWithMoreThanTwoValuesShouldThrowException3() throws Exception {
+    hash.apply(Arrays.asList("1", "2", "3"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void argumentListWithMoreThanTwoValuesShouldThrowException4() throws Exception {
+    hash.apply(Arrays.asList("1", "2", "3", "4"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void invalidAlgorithmArgumentShouldThrowException() throws Exception {
+    hash.apply(Arrays.asList("value to hash", "invalidAlgorithm"));
+  }
+
+  @Test
+  public void invalidNullAlgorithmArgumentShouldThrowException() throws Exception {
+    assertNull(hash.apply(Arrays.asList("value to hash", null)));
+  }
+
+  @Test
+  public void nullInputForValueToHashShouldProperlyThrowException() throws Exception {
+    assertNull(hash.apply(Arrays.asList(null, "md5")));
+  }
+
+  @Test
+  public void allAlgorithmsForMessageDigestShouldBeAbleToHash() throws Exception {
+    final String valueToHash = "My value to hash";
+    final Set<String> algorithms = Security.getAlgorithms("MessageDigest");
+
+    algorithms.forEach(algorithm -> {
+      try {
+        final MessageDigest m = MessageDigest.getInstance(algorithm);
+        m.update(valueToHash.getBytes(StandardCharsets.UTF_8));
+
+        assertEquals(BaseEncoding.base16().encode(m.digest()), hash.apply(Arrays.asList(valueToHash, algorithm)));
+      } catch (NoSuchAlgorithmException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  @Test
+  public void allAlgorithmsForMessageDigestShouldBeAbleToHashDirectStellarCall() throws Exception {
+    final String valueToHash = "My value to hash";
+    final Set<String> algorithms = Security.getAlgorithms("MessageDigest");
+
+    algorithms.forEach(algorithm -> {
+      try {
+        final Object actual = run("HASH('" + valueToHash + "', '" + algorithm + "')", Collections.emptyMap());
+
+        final MessageDigest m = MessageDigest.getInstance(algorithm);
+        m.update(valueToHash.getBytes(StandardCharsets.UTF_8));
+
+        assertEquals(BaseEncoding.base16().encode(m.digest()), actual);
+      } catch (NoSuchAlgorithmException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  @Test
+  public void nonStringValueThatIsSerializableShouldBeSerializedAndThenEncodedUsingHex() throws Exception {
```

Comment:
This is another nit, but I'll share the thought anyway:  A more typical (and shorter) name would be "testNonStringValueThatIsSerializable".  Being a "test", it is understood that the normative result is to properly do whatever it should do without error.  Similarly, the next two items could more briefly be named "testHashFunctionsWithVariablesAsInput" and "testStellarFunctionWhereOnlyHashTypeIsAVariable".

The last one, "aNonNullObjectThatDoesNotImplementSerializableShouldReturnAValueOfNull" is for a non-normative case, so it provides useful information as is.  It could still be shorter though.  How about "testNonSerializableObjectReturnsNull"?